source/core/NumericString.cpp

```diff
@@ -2118,9 +2127,13 @@ VIREO_FUNCTION_SIGNATURET(ByteArrayToString, ByteArrayToStringParamBlock)
         return _NextInstruction();
     TypedArrayCoreRef byteArray = _Param(ByteArrayIn);
     TypeRef eltType = byteArray->ElementType();
+    ConstCStr functionName = "Byte Array To String";
     if (!eltType->IsA(&TypeCommon::TypeInt8) && !eltType->IsA(&TypeCommon::TypeUInt8)) {
```

Comment:
This check should actually be a [hard error](https://github.com/ni/VireoSDK/blob/master/source/io/HttpClient.cpp#L160) which stops the runtime since we would never expect the editor to emit anything besides an Int8 or a UInt8.

I don't feel strongly about changing this from a runtime error since it is unlikely the editor would change types without this being caught. Do you agree or should this be changed to a hard error @sanmut?

Reply:
@rajsite It is ok to change this to a hard error. NXG before VIA generation must guarantee that input to this node is Int8 or UInt8. We should add tests in NXG to test arrays of different types such as Int16, UInt16, if we already don't have test coverage.

Reply:
I've changed this to a hard error in the third commit.  Any particular place the NXG array test should live?